jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1549,9 +1560,66 @@ Method getMethodHelper(
 			}
 		}
 	}
+	
+	/* The more specific method is retrieved via the top superclass rather than the direct interface
+	 * of this Class only if the method is also declared by another interface(retrieved via the top superclass)
+	 * which is not the same or super-interface of the direct interface of this Class.
+	 */
+	if (!forDeclaredMethod && (resultFromSuperClass != null)) {
+		candidateFromTopSuperClass = getMostSpecificMethodFromSuperClass(resultFromSuperClass, name, strSig, parameterTypes);
```

Comment:
Does this method look for only super classes and exclude current class?

Reply:
The code block here  is used compare with the method implemented by the current class only when the method implemented by one of superclases exists;
otherwise, the bestCandidate is determined by the method implemented by the current class:
```
	bestCandidate = result;
	Class<?> declaringClass = forDeclaredMethod ? this : result.getDeclaringClass();
	while (true) {
		result = declaringClass.getDeclaredMethodImpl(name, parameterTypes, strSig, result);
		if (result == null) {
			break;
		}
		boolean publicMethod = ((result.getModifiers() & Modifier.PUBLIC) != 0);
		if ((methodList != null) && publicMethod) {
			methodList.add(result);
		}
		if (forDeclaredMethod || publicMethod) {
			// bestCandidate and result have same declaringClass.
			Class<?> candidateRetType = bestCandidate.getReturnType();
			Class<?> resultRetType = result.getReturnType();
			if ((candidateRetType != resultRetType) && candidateRetType.isAssignableFrom(resultRetType)) {
				bestCandidate = result;
			}
		}
	}
```